### ISSUE TITLE
validation not always updating

### ISSUE THREAD

**Issue by thamera** [MrWolfZ/ngrx-forms#230]
Not sure if this is a bug or just my own mistake.  I am trying to get validations to work but they only seem to fire on the touched action, not my initial data load action (I preload some values).  Then the validations only show when individual fields are touched until the form is submitted.  Even after submitting, a field needs to be touched and then the validations show.

My form that is initialized as disabled:
```
const initialState = disable(updateRecursive(
  createFormGroupState<FormValue>(
    FORM_ID,
    {
      ticketRequester: '',
      ticketPriority: '',
      // ...list of fields shortened for brevity
    }
  ),
  s => setUserDefinedProperty(s, 'wasDisabled', s.isDisabled)
  )
);
```
I then load data from a service in my controller's on-init and feed in into an action
```
export class SetInitialValueAction implements Action {
  static readonly TYPE = `${FORM_ID} SET_INITIAL_VALUE`;
  readonly type = SetInitialValueAction.TYPE;
  constructor(public initialValue: FormValue) { }
}
```
action processed in reducer (with validator):
```
const validationFormGroupReducer = createFormStateReducerWithUpdate<FormValue>(updateGroup<FormValue>({
  ticketRequester: validate(required, email),
  ticketPriority: validate(required),
  ticketType: validate(required),
  // ticketDeadline: string;
  // ticketComplexity: string;
  ticketProject: validate(required),
  // ticketUpdateExisting: boolean;
  // ticketCC: string;
  ticketSubject: validate(required),
  ticketDescription: validate(required),
  // ticketFiles: any[];
}));

const reducers = combineReducers<State['freshdesk_submit'], any>({
  formState(state = initialState, a: Action): FormGroupState<FormValue> {
    let form = validationFormGroupReducer(state, a);

    switch (a.type) {
      case `${FORM_ID} SET_INITIAL_VALUE`:
        const payload = (a as SetInitialValueAction).initialValue;
        form = updateRecursive(
            enable(form),
            s => s.userDefinedProperties.wasDisabled ? disable(s) : s,
        );
        const updatedForm = setValue({
          ...form.value,
          ...payload
        })(form);
        console.dir(validationFormGroupReducer(updatedForm, a));
        return validationFormGroupReducer(updatedForm, a);

      default: {
        return form;
      }
    }
  },
});

export function reducer(s: State['freshdesk_submit'], a: Action): State['freshdesk_submit'] {
  return reducers(s, a);
}
```

Since I am calling the "ValidationFormGroupReducer" in the return on the SET_INITIAL_VALUE action, shouldn't it be coming back validated?  You can see I added a console log of the function before returning it and the errors object in the state is still empty at that point.  Error only show if I then touch/dirty a field.  Am I missing something here to get the validations to work (you can see I even run the validation on the state before catching the initial value action just to see if that would work.

**Comment by thamera** [comment]
I'm getting a little deeper on this  Helped by moving my reducers into the Angular 8+ approach using (e.g. using "createReducer" function) which makes these a little more readable for me).  I made a couple improvements but still trying to be able to get this to validate earlier.  So my current process overview is as follows:

1) route navigated
2) store setup with initial value (wrapped in "Disable" to disable the form until data load)
3) component on-init:  Load data and dispatch as payload on "Initial_Form" action I created that  does an "updateRecursive" to enable the form in the store and then "setValue" to apply the initial data values on the form
4) user interacts with form

I am trying to get validation to happen on step 3 above so that the user can immediately see the minimum fields they need to supply values for or fix but the "error" object in the store doesn't get populate until after the user starts interacting with the form at step 4.  Am I missing a potential function to run in step 3 to get the validations to run?

**Comment by thamera** [comment]
Going to go ahead and close this.  I found a couple key takeaways to get good validation early and often:

1.  use more recent NGRX approach (NGRX 8+) with constructs such as `createReducer()` and `createAction()`  Not an NGRX-FORMS items but rather a good general NGRX approach.  the reducers, actions and effects are soo much easier to read and build.  No exceptions here with this library.  Once I switched to that format, I could really see where WolfZ is going with this and it make a lot more sense now.  Nice job WolfZ!
2. Use `validationFormGroupReducer()` often!  great function.  I found it best to just have the final return of state in every reducer function to be wrapped in this function.  keeps it simple, and helps with the "validate often" approach.  When used with a clean display strategy like material, helps in giving clear indicators to the users.

I'm finding it useful within my reducer to tap into a combination of the built in actions (e.g. `onNgrxFormsAction(SetValueAction...` and my own custom actions (which again become easier to build in when using the newer NGRX 8+.  Nice job in the validation error approach.  Work really well for good reporting.  Now I just need to write some custom validators!
